src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -692,6 +692,7 @@ private void MakeCollectionExpressionTypeInferences(
                         MakeSpreadElementTypeInferences(spread, targetElementType, ref useSiteInfo);
                         break;
                     case BoundKeyValuePairElement keyValuePairElement:
+                    case BoundKeyValuePairExpressionElement keyValuePairExpressionElement:
                         // https://github.com/dotnet/roslyn/issues/77873: Handle input type inference for key:value elements.
```

Comment:
PROTOTYPE?

Reply:
https://github.com/dotnet/roslyn/issues/77873 is intended to cover the type inference changes needed.